demo/server/src/app.ts

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+import * as express from 'express';
+import * as path from 'path';
+import * as logger from 'morgan';
+import * as cookieParser from 'cookie-parser';
+import * as bodyParser from 'body-parser';
+import index from './routes/index';
+import * as favicon from 'serve-favicon';
+
+const app: express.Express = express();
+
+app.use(logger('dev'));
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(cookieParser());
+app.use(express.static(path.join(__dirname, 'public')));
+app.use(favicon(path.join(__dirname, 'public', 'favicon.ico')))
+
+app.use('/', index);
+
+//catch 404 and forward to error handler
+app.use((req: express.Request, res: express.Response, next) => {
+  var err: any = new Error('Not Found');
+  err.status = 404;
+  next(err);
+});
+
+//error handlers
+
+//development error handler
+//will print stacktrace
+if (process.env.NODE_ENV === 'development') {
+  app.use((err: any, req: express.Request, res: express.Response) => {
```

Comment:
For error handlers, though, we need a 4-arity function. Ignore unused param linting error if there's one.

If we declare a `express.ErrorHandler` variable it's easier to set it up with the if